### ISSUE TITLE
blake3 enables architecture-dependent features

### ISSUE THREAD

**Issue by vagrantc** [librestack/librecast#36]
Seems like blake3 passes compiler flags only present on x86_64 (maybe x86 too):

https://buildd.debian.org/status/package.php?p=librecast&suite=experimental

On arm64:

make[3]: *** Waiting for unfinished jobs....
cc: error: unrecognized command-line option ‘-mavx2’
cc: error: unrecognized command-line option ‘-msse4.1’
make[3]: *** [Makefile:49: blake3_avx2.o] Error 1
make[3]: *** [Makefile:46: blake3_sse41.o] Error 1
cc: error: unrecognized command-line option ‘-mavx512f’
cc: error: unrecognized command-line option ‘-mavx512vl’
make[3]: *** [Makefile:52: blake3_avx512.o] Error 1
make[3]: Leaving directory '/<<PKGBUILDDIR>>/blake3/c'
make[2]: *** [Makefile:14: blake3] Error 2
make[2]: Leaving directory '/<<PKGBUILDDIR>>/libs'
make[1]: *** [Makefile:24: libs] Error 2
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_build: error: make -j8 "INSTALL=install --strip-program=true" returned exit code 2
make: *** [debian/rules:4: binary-arch] Error 25

The full build log for arm64:

https://buildd.debian.org/status/fetch.php?pkg=librecast&arch=arm64&ver=0.4.2-1&stamp=1615241703&raw=0

**Comment by vagrantc** [comment]
Will work around this for now by switching to libsodium for non-x86.

**Comment by brettsheffield** [comment]
Yeah, those flags are x86_64. There's a portable version we can use for non-x86
